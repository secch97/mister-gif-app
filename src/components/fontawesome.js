//Import the library
import {library} from  "@fortawesome/fontawesome-svg-core";

//Import your icons
import { faCircleUp, faSpinner, faFaceSadCry, faLink } from "@fortawesome/free-solid-svg-icons";

library.add(
    faCircleUp,
    faSpinner,
    faFaceSadCry,
    faLink
);
