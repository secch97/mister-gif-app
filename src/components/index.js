export * from "./NavigationBar"
export * from "./AddCategory";
export * from "./GifGrid";
export * from "./GifItem";
export * from "./GifNotFound";
