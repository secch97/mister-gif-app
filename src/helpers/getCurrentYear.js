const getCurrentYear = () => {
    const currentDate = new Date();
    const currentYear = currentDate.getFullYear();
    return currentYear.toString();
};

export {
    getCurrentYear
};