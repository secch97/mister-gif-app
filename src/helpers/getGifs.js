const getGifs = async(category) => {
    const url = `https://api.giphy.com/v1/gifs/search?api_key=FWCszwwyG9YJM5R3VACQkGLUQBl2M7Qe&q=${category}&limit=10`;
    const response = await fetch(url);
    const {data} = await response.json();
    const gifs = data.map((gif) => {
        return {
            id: gif.id,
            title: gif.title,
            url: gif.images.downsized_medium.url
        };
    });
    return gifs;
};

export {
    getGifs
};