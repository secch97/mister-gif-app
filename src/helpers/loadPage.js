const loadPage = (time, setAppLoaded) => {
     setTimeout(() => {
        setAppLoaded(true);
    }, time);
    return;
};

export {
    loadPage
};