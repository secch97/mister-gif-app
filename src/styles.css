* {
    font-family: Helvetica, Arial, sans-serif;
  }
  
  body {
    padding: 60px;
    background-color: rgb(234, 234, 234);

  }
  form {
    display: flex;
    flex-direction: column;
    flex: 1 1 auto;
    justify-content: center;
    align-items: center;
    width: 100%;
  }

  input {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 50%;
  }
  
  h2 {
    font-size: 1.5rem;
  }
  
  h3 {
    font-size: 3rem;
    margin-bottom: 5px;
  }



  #root{
    display:flex;
    flex-direction: column;
    flex:1;
    justify-content: center;
    align-items: center;
  }
  
  .card-grid-container{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
   
  }
  
  .card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }


  
  .card {
    background-color: white;
    border-radius: 10px;
    box-shadow: 10px 10px 0px rgba(0, 0, 0, 0.251);
    display: flex;
    flex-direction: column;
 

    width: 380px;
    height: 250px;  

    margin-bottom: 30px;
    margin-right: 40px;
    overflow: hidden;

    position: relative;

  }
  
  .card span {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-end;
   
    width: inherit;
    height: inherit;
    color:transparent;
    flex: 1;
    font-size: 1.5rem;
    text-align: center;
    position: absolute;
    margin: 0px;

    text-transform: capitalize;
  }
  
  .card img {
    width: 100%;
    height: 100%;
  }

  .card:hover{
    transform: scale(1.05);
    transition: ease-in-out 0.15s;
    box-shadow: 10px 10px 0px rgba(233, 69, 96, 0.251);
  }

  .card span:hover{
    color: white;
    text-shadow: 2px 2px #E94560;
  }