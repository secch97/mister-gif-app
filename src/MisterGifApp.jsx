import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { NavigationBar, GifGrid } from './components/';

import {useState} from 'react';


const MisterGifApp = () => {

    const [categories, setCategories] = useState(["Trending"]);
    const [scroll, setScroll] = useState(0);

    const onAddCategory = (newCategory) => {
        setCategories([newCategory, ...categories.filter((category) => category.toLowerCase()!==newCategory.toLowerCase())]);
    };

    const scrollToTop = () => {
        window.scrollTo({ top: 0, behavior: 'smooth' });
    }

    return (
        <>
            <NavigationBar onAddCategory={onAddCategory} ></NavigationBar>
            {
                categories.map((category, i) => {
                    return (
                        <GifGrid 
                            key={`${category}`}
                            category={category}
                            firstPosition={!i ? true:false}
                        />
                    );
                    })
            }
            <a className='button-top' onClick={scrollToTop}>
                <FontAwesomeIcon 
                icon={["fas", "circle-up"]} 
                size="4x"
                />
            </a>
        </>
  );
};

export {
    MisterGifApp
};
