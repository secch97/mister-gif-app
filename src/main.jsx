import React, { useState } from 'react';
import ReactDOM from 'react-dom/client';
import { MisterGifApp } from './MisterGifApp';
import "./styles.css";
import './components/fontawesome';

ReactDOM.createRoot(document.getElementById('root')).render(
        <MisterGifApp />
);
