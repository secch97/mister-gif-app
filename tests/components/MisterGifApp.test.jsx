import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { MisterGifApp } from "../../src/MisterGifApp";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import "./fontawesome";
import { loadPage } from "../../src/helpers/loadPage";
import { act } from "react-dom/test-utils";
import { scrollToTop } from "../../src/helpers/scrollToTop";

describe('MisterGifApp testing', () => { 
    window.scrollTo = jest.fn();

     test('Should match snapshot', () => { 
        const {container} = render(<MisterGifApp></MisterGifApp>);
        expect(container).toMatchSnapshot();
      });

     test('Should stop loading after 1s', () => { 
        jest.useFakeTimers();
        render(<MisterGifApp/>);
        act(()=>jest.advanceTimersByTime(1000));
        expect(screen.getByText("Trending")).toBeTruthy();
      });

      test('Should add a new category when submitting form', () => { 
        jest.useFakeTimers();
        render(<MisterGifApp/>);
        act(()=>jest.advanceTimersByTime(1000));
        const input = screen.getByRole("textbox");
        const form = screen.getByRole("form");
        fireEvent.input(input, {target:{value:"Mr Robot"}});
        fireEvent.submit(form);
        expect(screen.getAllByRole("heading", {level: 1}).length).toBe(2);
        expect(screen.getByText("Mr Robot")).toBeTruthy();
        window.scrollTo.mockClear();
      });

      test('Should add a new category when clicking the button', () => { 
        jest.useFakeTimers();
        render(<MisterGifApp/>);
        act( () => jest.advanceTimersByTime(1000));
        const input = screen.getByRole("textbox");
        const button = screen.getByRole("button");
        console.log(button);
        fireEvent.input(input, {target:{value: "Mr Robot"}})
        fireEvent.click(button);
        screen.debug();
        expect(screen.getAllByRole("heading", {level:1}).length).toBe(2);
        expect(screen.getByText("Mr Robot")).toBeTruthy();
    });
 });