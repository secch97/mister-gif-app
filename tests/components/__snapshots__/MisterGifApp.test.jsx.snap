// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MisterGifApp testing Should match snapshot 1`] = `
<div>
  <div
    class="mister-gif"
  >
    <div
      class="loading"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-spinner fa-spin fa-6x "
        data-icon="spinner"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zm0 416c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zM48 304c26.5 0 48-21.5 48-48s-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48zm464-48c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zM142.9 437c18.7-18.7 18.7-49.1 0-67.9s-49.1-18.7-67.9 0s-18.7 49.1 0 67.9s49.1 18.7 67.9 0zm0-294.2c18.7-18.7 18.7-49.1 0-67.9S93.7 56.2 75 75s-18.7 49.1 0 67.9s49.1 18.7 67.9 0zM369.1 437c18.7 18.7 49.1 18.7 67.9 0s18.7-49.1 0-67.9s-49.1-18.7-67.9 0s-18.7 49.1 0 67.9z"
          fill="currentColor"
        />
      </svg>
      <h2>
        Loading
      </h2>
    </div>
  </div>
</div>
`;
